src/datamodel_code_generator/parser/jsonschema.py

```
@@ -9714,6 +9802,10 @@ def _get_ref_body_from_local_http_path(self, ref: str) -> dict[str, YamlValue]:
         msg = f"$ref local file not found for {ref}: tried {', '.join(str(path) for path in file_paths)}"
         raise Error(msg)
 
+    def _is_directory_read_error(self, exc: OSError, path: Path) -> bool:  # noqa: PLR6301
```

Ruff (0.16.3)

[warning] `noqa` comment used instead of `ruff: ignore`

Use `ruff: ignore` instead

(RUF105)